src/object/object_primitive.c

```diff
@@ -11212,12 +11307,20 @@ mr_readval_string_internal (OR_BUF * buf, DB_VALUE * value, TP_DOMAIN * domain,
 		  start = buf->ptr;
 		}		/* size != -1 */
 
-	      str_length = or_get_varchar_length (buf, &rc);
+	      /* Get the length of the string, be it compressed or uncompressed. */
+	      rc = or_get_varchar_compression_lengths (buf, &compressed_size, &decompressed_size);
 	      if (rc != NO_ERROR)
 		{
 		  return ER_FAILED;
 		}
-
+	      if (compressed_size <= 0)
+		{
+		  str_length = decompressed_size;
+		}
+	      else
+		{
+		  str_length = compressed_size;
+		}
```

Comment:
@lucianlazar2211 

Would you explain the reason you removed the assertion?
If it does not set, code may not work.